servers/audio/audio_stream.cpp

```diff
@@ -76,6 +76,10 @@ void AudioStreamPlaybackResampled::mix(AudioFrame *p_buffer, float p_rate_scale,
 			internal_buffer[1] = internal_buffer[INTERNAL_BUFFER_LEN + 1];
 			internal_buffer[2] = internal_buffer[INTERNAL_BUFFER_LEN + 2];
 			internal_buffer[3] = internal_buffer[INTERNAL_BUFFER_LEN + 3];
+			if(!is_playing()) {
+				memset(internal_buffer + 4, 0, INTERNAL_BUFFER_LEN);
```

Comment:
As a general rule, system calls should not be used in audio code as, depending on implementation, they may cause scheduling. Just use a regular loop I guess..

Reply:
memset() is not generally implemented in the kernel though, I can switch it out with a loop but on most compilers this will end up with some sse optimized loop straight into the binary. Anything I do here is likely going to be slower than relying on the compiler to do something clever.

I'll change it if you meant any library call, but if you meant 'system call' as in 'a call into the kernel' then this should be fine.

Reply:
I can wrap the memset into an #ifdef __GNUC__ and do a loop on msvc? clang/gcc have documented that memset gets inlined by the compiler. I can't find such a statement about msvc :-/